listenbrainz/webserver/static/js/src/user/Listens.tsx

```diff
@@ -792,7 +791,7 @@ export default class Listens extends React.Component<
     const isOlderButtonDisabled =
       !nextListenTs || nextListenTs <= oldestListenTs;
     const isOldestButtonDisabled =
-      listens?.[listens?.length - 1]?.listened_at <= oldestListenTs;
+      listens?.[(listens?.length ?? 0) - 1]?.listened_at <= oldestListenTs;
```

Comment:
Yes, I think I took a shortcut when I was correcting this.
My reasoning was that in that case `listens[-1]` will be undefined and with the optional chaining (`?.`) this will be the same as an undefined `listens[0]`.
But perhaps I should rewrite it so it's clearer, there's a lot of assumption in all that.